gum/backend-x86/gumstalker-x86.c

```diff
@@ -2601,9 +2608,52 @@ gum_exec_block_backpatch_inline_cache (GumExecBlock * block,
   }
 }
 
+
+#if GLIB_SIZEOF_VOID_P == 4 && !defined (HAVE_QNX) && defined (HAVE_WINDOWS)
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_wow64transition (GumExecBlock * block,
+                                           GumGeneratorContext * gc,
+                                           GumExecCtx *ec)
+{
+  GumX86Writer * cw = gc->code_writer;
+  guint8 code[] = {
+    /* 0x00 */ 0x50,                                           /* push eax */
+    /* 0x01 */ 0x8b, 0x44, 0x24, 0x04,                         /* mov eax, dword [esp + 4] */
+    /* 0x05 */ 0x89, 0x05, 0xaa, 0xaa, 0xaa, 0xaa,             /* mov dword [0xaaaaaaaa], eax */
+    /* 0x0b */ 0xc7, 0x44, 0x24, 0x04, 0xbb, 0xbb, 0xbb, 0xbb, /* mov dword [esp + 4], 0xbbbbbbbb */
+    /* 0x13 */ 0x58,                                           /* pop eax */
+    /* 0x14 */ 0xff, 0x25, 0xcc, 0xcc, 0xcc, 0xcc,             /* jmp dword [0xcccccccc] */
+    /* 0x1a */ 0x90, 0x90, 0x90, 0x90                          /* <saved ret-addr here> */
+  };
+  const gsize store_ret_addr_offset = 0x05 + 2;
+  const gsize load_continuation_addr_offset = 0x0b + 4;
+  const gsize wow64transition_addr_offset = 0x14 + 2;
+  const gsize saved_ret_addr_offset = 0x1a;
+
+  gconstpointer resolve_dynamically_label = cw->code;
+
+  gum_exec_block_close_prolog (block, gc);
+
+  gpointer * saved_ret_addr = (gpointer *) (cw->code + saved_ret_addr_offset);
+  gpointer continuation = cw->code + saved_ret_addr_offset + 4;
+
+  *((gpointer *) (code + store_ret_addr_offset)) = saved_ret_addr;
+  *((gpointer *) (code + load_continuation_addr_offset)) = continuation;
+  *((gpointer *) (code + wow64transition_addr_offset)) = ec->stalker->wow64transition_address;
+
+  gum_x86_writer_put_bytes (cw, code, sizeof (code));
+
+  gum_exec_block_write_sysenter_continuation_code(block, gc, saved_ret_addr, resolve_dynamically_label);
+
+  return GUM_REQUIRE_NOTHING;
+}
+#endif
+
+
 static GumVirtualizationRequirements
 gum_exec_block_virtualize_branch_insn (GumExecBlock * block,
-                                       GumGeneratorContext * gc)
+                                       GumGeneratorContext * gc,
+                                       GumExecCtx * ec)
```

Comment:
The `block` already has a pointer to the `GumExecCtx` in its `ctx` field, so the additional parameter is not needed.